packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -268,16 +268,23 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
         return ids as Set<string>;
     }
 
-    public async getBlob(blobId: string): Promise<ArrayBufferLike> {
+    //* Add param for allowTombstone header
+    public async getBlob(blobId: string, allowTombstone: boolean = false): Promise<ArrayBufferLike> {
         const request = { url: blobId };
         if (this.tombstonedBlobs.has(blobId) ) {
+            const shouldFail = this.throwOnTombstoneUsage && !allowTombstone;
             const error = responseToException(createResponseError(404, "Blob removed by gc", request), request);
-            const event = {
-                eventName: "GC_Tombstone_Blob_Requested",
-                url: request.url,
-            };
-            sendGCTombstoneEvent(this.mc, event, this.runtime.clientDetails.type === summarizerClientType, [BlobManager.basePath], error);
-            if (this.throwOnTombstoneUsage) {
+            sendGCTombstoneEvent(
+                this.mc.logger,
+                {
+                    eventName: "GC_Tombstone_Blob_Requested",
+                },
+                shouldFail /* logAsError */,
+                this.runtime.clientDetails.type === summarizerClientType,
+                [BlobManager.basePath],
+                error,
+            );
+            if (shouldFail) {
                 throw error;
```

Comment:
This error will be returned by handle.get(). We need to make sure that consumers can look at it and know that this is caused by tombstone and re-try with the header.

Reply:
I'm gonna revert blob stuff for now.  Good catch though

Reply:
Tombstone errors thrown to (through) handle.get will include a headers object like `{ isTombstoned: true }`